app/views/admin/products_v3/_variant_row.html.haml

```diff
@@ -9,13 +9,13 @@
   = error_message_on variant, :sku
 %td
   -# empty
-- if variant.persisted?
-  %td.align-right
-    .content= variant.unit_to_display
-- else # until unit component is developed, use a basic input just so we can create new records
-  %td.field
-    = f.number_field :unit_value, 'aria-label': t('admin.products_page.columns.unit')
-    = error_message_on variant, :unit_value
+%td.field
+  = f.text_field :unit_value,
+      value: number_with_precision(variant.unit_value, precision: nil,strip_insignificant_zeros: true),
+      'aria-label': t('admin.products_page.columns.unit_value')
```

Comment:
> Unfortunately we can't use an input[type=number] because the desired step will vary. (eg sometimes you want a decimal point, but other times not).

Is that really a blocker? The step only affects the up and down controls. You can still enter any value. I think that the default step of 1 is okay for most cases and if someone handles 3.75 kg then they just can't use the step controls. They don't really work that well in that case anyway because the we would need to use step 0.001 and then it would take ages to adjust the field that way.

Reply:
Good point.
But I later found out that this input may also handle text input (it allows input for both `unit_value` and `unit_description`). Sorry I didn't go back and update this commit.